### ISSUE TITLE
zipkin ui refresh request again response no anything

### ISSUE THREAD

**Issue by weixsun** [openzipkin/zipkin#2443]
docker-zipkin version:2.12.5

docker-zipkin version : 2.11.12 is ok. other version not test

deploy env: kubernetes v1.13.1

simple deploy yaml:
```yaml
env:
            - name: "STORAGE_TYPE"
              value: "elasticsearch"
            - name: "ES_HOSTS"
              value: "http://192.168.66.62:9200"
            - name: "ES_MAX_REQUESTS"
              value: "64"
            - name: "ES_INDEX"
              value: "rdc_cloud_zipkin_trace"
            - name: "ElasticsearchStorage.Builder.index"
              value: "rdc_cloud_zipkin_trace"
            - name: "ES_DATE_SEPARATOR"
              value: "."
            - name: "ElasticsearchStorage.Builder.dateSeparator"
              value: "."
            - name: "ES_INDEX_SHARDS"
              value: "5"
            - name: "ES_INDEX_REPLICAS"
              value: "1"
            - name: "KAFKA_BOOTSTRAP_SERVERS"
              value: "192.168.67.21:9092,192.168.67.22:9092,192.168.67.23:9092"
            - name: "KAFKA_TOPIC"
              value: "rdc_cloud_zipkin_trace"
            - name: "KAFKA_GROUP_ID"
              value: "rdc_cloud_zipkin_trace_zipkin_group"
            - name: "QUERY_PORT"
              value: "9411"
            - name: "ZIPKIN_UI_DEPENDENCY_LOW_ERROR_RATE"
              value: "0.3"
            - name: "ZIPKIN_UI_DEPENDENCY_HIGH_ERROR_RATE"
              value: "0.75"
            - name: "ZIPKIN_UI_LOGS_URL"
              value: "http://devcloud.i139.cn/kibana/app/kibana#/discover?_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(_source),index:'b7e371c0-4638-11e9-bad2-61d86dd48551',interval:auto,query:(language:lucene,query:'{traceId}'),sort:!(timestamp_millis,desc))"
            - name: "JAVA_OPTS"
              value: "-Xms1G -Xmx1G -Dlogging.level.zipkin=WARN -Dlogging.level.zipkin2=WARN"
```

First request zipkin ui every thing is OK.

![1111552621436_ pic_hd](https://user-images.githubusercontent.com/29357333/54407146-a3d55d80-4717-11e9-990c-e16d7d1b6415.jpg)

But when refresh zipkin ui in chrome browser, chrome browser print no thing,however the response is 200, And use curl http://zipkin-ui-url:9411 response normal data forever:

![1121552621504_ pic_hd](https://user-images.githubusercontent.com/29357333/54407188-cf584800-4717-11e9-8f34-2a4ec218cf89.jpg)

![1101552621381_ pic_hd](https://user-images.githubusercontent.com/29357333/54407101-82747180-4717-11e9-97ae-f6d24f8b09fe.jpg)

**Comment by codefromthecrypt** [comment]
cc @anuraaga @trustin the http 200 with zero data sounds dubious but not sure if that is directly from the server or from k8s.

I will look into this tonight. hopefully can be reproducable outside k8s if only docker.

If anyone has armeria troubleshooting tips for dubious http responses.. like log categories etc please pass them.

**Comment by msmsimondean** [comment]
I'm seeing this issue too.  I see it both with Zipkin running locally on Docker and also with Spring Sleuth's demo instance https://docssleuth-zipkin-server.cfapps.io/zipkin/

**Comment by msmsimondean** [comment]
The issue seems to start in version `2.12.4`.  It affects both `2.12.4` and `2.12.5`.  Rolling back to version `2.12.3` of the Docker image fixes the issue for me.

**Comment by codefromthecrypt** [comment]
the main change was switch to armeria library for http routing. I maybe
created a bug here.

I will investigate tonight and thanks for checking. it is good that it is
reproducible outside k8s

**Comment by codefromthecrypt** [comment]
verified I get the white screen using latest zipkin (built from master) just by clicking refresh cc @tacigar

**Comment by codefromthecrypt** [comment]
it doesn't matter whether it is classic or lens btw

**Comment by codefromthecrypt** [comment]
the workaround right now is to do force reload. I'm looking into this.

**Comment by codefromthecrypt** [comment]
There is a bug in cache control handling. The server should be returning 304, but it is returning 200

Ex.
```bash
$ curl -H'If-None-Match: "7mC/RgQ/AWmBfiYQ"' localhost:9411/zipkin/ -v
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 9411 (#0)
> GET /zipkin/ HTTP/1.1
> Host: localhost:9411
> User-Agent: curl/7.54.0
> Accept: */*
> If-None-Match: "7mC/RgQ/AWmBfiYQ"
> 
< HTTP/1.1 200 OK
< date: Fri, 15 Mar 2019 13:21:35 GMT
< last-modified: Fri, 15 Mar 2019 13:15:22 GMT
< etag: "7mC/RgQ/AWmBfiYQ"
< content-type: text/html
< cache-control: max-age=60
< content-length: 0
< 
* Connection #0 to host localhost left intact
```

**Comment by codefromthecrypt** [comment]
I think I found the bug

**Comment by codefromthecrypt** [comment]
https://github.com/openzipkin/zipkin/pull/2445 will fix this

**Comment by codefromthecrypt** [comment]
2.12.6 on the way. thanks for the bug reports and sorry I made this bug

**Comment by codefromthecrypt** [comment]
ok I verified it is good with docker now

**Comment by msmsimondean** [comment]
Thanks @adriancole.  The fix in `2.12.6` is working for me.

**Comment by codefromthecrypt** [comment]
good to hear @msmsimondean !
